docs/reference/token-types.md

```diff
@@ -0,0 +1,168 @@
+it verbatim in the request to the server (e.g. in the `Authorization` header). These are the simplest types of token
+for clients to use, but allow for interception and replay attacks. In particular, the server that receives a Bearer Token
+is capable of extracting it and using it to perform other API calls.
+
+A ***Hawk Token*** consists of an opaque string id and corresponding secret key. To use the token, clients must use the
+secret key to sign outgoing requests using [the scheme described here](https://hapi.dev/family/hawk/). This is more
+complicated for clients but it binds the use of the token to the particular request being made, and it offers the
+potential for replay prevention.
+
+An ***FxA Hawk Token*** is a 32-byte string from which the id and key of a ***Hawk Token*** can be generated using HKDF.
+This simplifies client handling of the token since it only has to persist a single value.
+
+# Session Tokens
+
+A `sessionToken` is the Firefox Accounts equivalent of traditional website login session cookie - it represents
+the user who is currently signed in to accounts.firefox.com, and it used to authenticate many calls from the
+FxA front-end website to the FxA back-end API.
+
+Session tokens are an ***FxA Hawk Token*** where the 32-byte string is generated by the server. They can be obtained by:
+* The successful creation of a new account, via the `/account/create` endpoint.
+* A successful login to a new account, via the `/account/login` endpoint.
+* Duplicating an existing `sessionToken`, via the `/session/duplicate` endpoint.
+* Providing the id of a previous `sessionToken` when changing the account password, via the `/password/change/finish`
+  endpoint.
+* Requesting the `https://identity.mozilla.com/tokens/session` scope during an OAuth flow.
+
+Session tokens are intended only for authenticated communication with the FxA servers themselves, so RP servers
+should never encounter a session token in the wild.
+
+On the client side, session tokens are typically handled by web content on accounts.firefox.com while the user
+is interacting with their account. However, web browsers that support signing in with FxA in order to access
+account-enabled browser features, may obtain a session token during the sigin flow and use it to communicate
+directly with the FxA server APIs.
+
+# Session Metadata
+
+Since session tokens represent a signed-in user, the server maintains a lot of additional metadata about the user's
+signed-in state. Properties of the session token include:
+
+* The types of authentication performed by the user, and level of confidence in their identity.
+* Timestamp at which authentication last occurred (which can be increased by prompting the user to re-enter their
+  password).
+* Timestamp at which the token was last used (truncated to a few hours granularity to reduce db write load).
+* Details of the user-agent string from the last use of the token.
```

Comment:
The user's ip based geo location is also on the session token.

Reply:
added to new version